neural_compressor/adaptor/torch_utils/gptq.py

```diff
@@ -0,0 +1,642 @@
+    Args:
+        module: torch.nn.Module
+        module_types: List of torch.nn.Module.
+
+    Returns:
+        gptq_related_blocks = {
+            "embeddings": {}, # Dict embedding layers before transfromer stack module, 
+            "transformers_pre": {}, # TODO
+            "transformers_name": string. LLMs' transformer stack module name ,
+            "transformers": torch.nn.ModuleList. LLMs' transformer stack module,
+            "transformers": {}, Dict# TODO
+        }
+    """
+    gptq_related_blocks = {
+        "embeddings": {},
+        "transformers_pre": {}, # todo
+        #"transformers_name": None, # None
+        #"transformers": None, # None
+        "transformers_name": "", # None
+        "transformers": [], # None
+        "transformers_post": {}, # todo
+    }
+    for n, m in module.named_modules():
+        if type(m) in module_types:
+            gptq_related_blocks["transformers_name"] = n
+            gptq_related_blocks["transformers"] = m
+            return gptq_related_blocks
+        else:
+            if is_leaf(m):
+                gptq_related_blocks["embeddings"][n] = m
+    return gptq_related_blocks
+
+def find_layers(module, layers=[nn.Conv2d, nn.Conv1d, nn.Linear, transformers.Conv1D], name=''):
+    """Get all layers with target types."""
+    if type(module) in layers:
+        return {name: module}
+    res = {}
+    for name1, child in module.named_children():
+        res.update(find_layers(
+            child, layers=layers, name=name + '.' + name1 if name != '' else name1
+        ))
+    return res
+
+def find_layers_name(module, layers=[nn.Conv2d, nn.Conv1d, nn.Linear, transformers.Conv1D], name=''):
+    """Get all layers with target types."""
+    if type(module) in layers:
+        return [name]
+    res = []
+    for name1, child in module.named_children():
+        res += find_layers_name(child, layers=layers, name = name + '.' + name1 if name != '' else name1)
+    return res
+
+def log_quantizable_layers_per_transformer(
+        transformer_blocks, 
+        layers=[nn.Conv2d, nn.Conv1d, nn.Linear, transformers.Conv1D]
+    ):
+    """Print all layers which will be quantized in GPTQ algorithm."""
+    logger.info("* * Layer to be quantized * *")
+
+    for block_id in range(len(transformer_blocks['transformers'])):
+        transformer_block = transformer_blocks['transformers'][block_id]
+        layers_for_this_tblock = find_layers_name(transformer_block)
+        layer_names = [
+            (transformer_blocks['transformers_name'] + "." + str(block_id) + '.' + layer_name) \
+            for layer_name in layers_for_this_tblock
+        ]
+        for name in layer_names:
+            logger.info(name)
+
+#===========================================
+
+#===============quantization related============================
+def quantize(x, scale, zero, maxq):
+    """Do quantization."""
+    if maxq < 0:
+        return (x > scale / 2).float() * scale + (x < zero / 2).float() * zero

```

Comment:
better call the RTN quantization function directly in the future.  When will the maxq <0? If it is needed, you can change rtn quantization function accordingly.